systemd/systemd_test.go

```diff
@@ -236,3 +244,98 @@ func TestAddCPUQuota(t *testing.T) {
 		})
 	}
 }
+
+func TestOOMPolicyApply(t *testing.T) {
+	if !IsRunningSystemd() {
+		t.Skip("Test requires systemd.")
+	}
+	if !cgroups.IsCgroup2UnifiedMode() {
+		t.Skip("cgroup v2 is required")
+	}
+	if os.Geteuid() != 0 {
+		t.Skip("Test requires root.")
+	}
```

Comment:
nit: do you need to add a check for systemd version here, to skip the test for older versions?

Or, perhaps, get the unit properties, check if the `OOMPolicy` is there (if it's always there on newer systemd, of course), otherwise `t.Skip("no OOMPolicy property; older systemd?")`